common/models/transducer/topology.py

```diff
@@ -0,0 +1,50 @@
+from typing import Callable, List
+
+from returnn.tf.util.data import Data
+from returnn.tf.layers.base import LayerBase
+from tensorflow import Tensor
+
+from .loss import (rnnt_loss, rnnt_loss_out_type,
+                   rnnt_tf_loss, rna_tf_loss, rna_loss_out_type)
+from .alignment import (rnnt_alignment, rnnt_alignment_out_type,
+                        rna_alignment, rna_alignment_out_type, SourceType)
+
+
+class Topology:
+  """
+  Class to hold informations about different topologies such as loss-, alignment-funcion and their Data_out_types.
+  loss and alignment functions are to be used in eval like layers that return a source function.
+  """
+  def __init__(self, name: str,
+               loss: Callable[[SourceType], Tensor],
+               loss_out_type: Callable[[], Data],
+               alignment: Callable[[SourceType], Tensor],
+               alignment_out_type: Callable[[List[LayerBase], ], Data]):
+    """ Topology such as rnnt, rna, ctc.
+    Args:
+        name (str): name of the alignment
+        loss (Callable[[SourceType], Tensor]): loss function
```

Comment:
```suggestion
        loss (((i: int, auto_convert: bool = True, enforce_batch_major: bool = False, as_data: bool = False)-> Union[Data,tf.Tensor]) -> Tensor): loss function
```

@albertz  Smth like this? We have to give the type of a callable that takes a callable as input param.

Reply:
Keep it simple.

```suggestion
        loss: function (source: (i: int, as_data: bool = False, ...) -> tf.Tensor|Data, self, ...) -> tf.Tensor
```